### DISCUSSION TITLE
Problem with screen on/off

### DISCUSSION THREAD

**Discussion by FabienArr** [tbnobody/OpenDTU#1906] [Q&A]
Hello,

I don't understand, i have problem with my screen which sometimes is on the night is off. But the morning he doesn't restart but i have the information on the webpage of my inverters.

I have this wiring : https://waldy-smarthome.de/opendtu/images/OpenDTU-Wiring-CMT2300A-Display.png

My profile is : 
![image](https://github.com/tbnobody/OpenDTU/assets/87081538/8bb6a9bb-0aac-410a-91c1-ee00c206c04e)

The config screen : 
![image](https://github.com/tbnobody/OpenDTU/assets/87081538/104943df-d9c2-41b8-9d20-3a5e05f945f0)

I need to restart or change the profile file and restart for the screen to display again, it's very curious.

Thanks for your help.

**Comment by ghost** [comment]


**Reply by FabienArr** [reply]
Hello,

Yes it is the correct display, i have a SSD1306 0,96", the screen works 2 days.
I have buy this kit https://www.ebay.fr/itm/186367887005 just with a SSD1306 more.
I will buy a new ESP32-WROOM-32 Board and a new screen screen a little bigger.
But it's curious, all works well during 2 days and now the screen is always off.

Regards

**Reply by ghost** [reply]
Buy here in the [SolarZeroShop](https://www.ebay.de/str/solarzeroshop).
You can also inquire before making your purchase. The seller has other items that are not listed in the shop.
Everything I ordered there works and the support is very good.

(And if I'm not mistaken, there is also a connection between the eBay shop and OpenDTU on Github.)

Otherwise I have no further ideas for your problem.
Something may also be defective.

**Reply by tbnobody** [reply]
> And if I'm not mistaken, there is also a connection between the eBay shop and OpenDTU on Github

Nope

**Reply by FabienArr** [reply]
It's complicated for me, i'm in France and in the most of cases the delivery tax are expansive.
His shop is empty on his website and on ebay.
All works fine but not the screen : 
![image](https://github.com/tbnobody/OpenDTU/assets/87081538/cf983095-c067-49a2-82ea-7b14fa20b5e6)

**Reply by FabienArr** [reply]
I use the [opendtu-generic.bin]

**Reply by ghost** [reply]
@ FabienArr 
I use the: opendtu-generic_esp32.bin

@tbnobody 
Dann habe ich mich wohl geirrt. 
Allerdings lesen die Jungs und Mädels vom Shop hier auf auf jeden Fall mit. Das stellte sich bei anfänglichen und gelösten Problemen mit meiner dort gekauften Hardware heraus.

**Reply by FabienArr** [reply]
@schneeer Do you think i need to use the esp32.bin ?
Because i have this 👍 
![image](https://github.com/tbnobody/OpenDTU/assets/87081538/6c9aec8d-1b16-4796-87f2-ed3b9fc60ae7)

**Reply by ghost** [reply]
Hi,
that is entirely possible.
I would have to read that up: but I think depending on the type and number of components connected to the ESP (only cmt300, or/and nrf24, and display...) it has to be an ESP32.
The xxx.bin file should be appropriate for the board being used.
It's definitely worth a try.

Salut,
c'est tout à fait possible.
Il faudrait que je lise cela : mais je pense qu'en fonction du type et du nombre de composants connectés à l'ESP (uniquement cmt300, ou/et nrf24, et écran...), il doit s'agir d'un ESP32.
Le fichier xxx.bin doit être approprié à la carte utilisée.
Cela vaut vraiment la peine d'essayer.
(By Google translated)

**Reply by FabienArr** [reply]
Ok thanks, it's the solution.
The seller tell me to use the generic not the generic for ESP32.
Many thanks.

**Reply by FabienArr** [reply]
Raah, it's very curious, after firmware update with the generic for ESP32, the display works, and after an afternoon later the screen display again nothing. I'm lost. SOrry, i thought it's was ok and now no.

**Reply by ghost** [reply]
I have another idea.
It could be the display.

The SSD-1306 is available in various versions with different board layouts.
Partially with an option to change the I2C address.

Out of my own interest, I looked into the software a bit. What surprises me about openDTU is that no I2C address has to be specified for the displays in the program code. The only exception is display type 5.

I've done one thing or another with the Arduino IDE, and the I2C address had to be specified each time. In any case, you need the I2C address if 2 displays are to be connected.
I also use the U8g2 library used by openDTU.
Maybe the openDTU developers or someone else can say something about this?

Here are 2 variants:
![1](https://github.com/tbnobody/OpenDTU/assets/155244148/80d7a627-1581-4374-8806-719679567ab1)
![2](https://github.com/tbnobody/OpenDTU/assets/155244148/53502279-f1c0-49b6-aa42-f14532b1a258)
I only buy the 1st variant, this definitely works for me with OpenDTU too. (That is, I actually only use LCDs whenever possible.) I have multiple OpenDTUs. The one that always runs has an SH1106.
Maybe your display is not 100% recognized/supported?

**Reply by ghost** [reply]
It won't help, there is newer firmware: [v24.4.12](https://github.com/tbnobody/OpenDTU/releases/tag/v24.4.12)
According to your screenshot you are using v24.3.31.

**Reply by FabienArr** [reply]
I have the last firmware now, when i update the screen works for an afternoon : 
![image](https://github.com/tbnobody/OpenDTU/assets/87081538/99b260ff-5417-478b-8aed-790d4d560fee)

This is my screen : 
![image](https://github.com/tbnobody/OpenDTU/assets/87081538/67a24ef7-8fe6-4df7-8d1c-a0b9437221f3)

It's not the good variant ?

But why it works for 2 hours or 3 hours and after not ?

Regards

**Reply by ghost** [reply]
Yes, the firmware is fine and so is the display.
0x78 is the "same" as 0x3C. This has something to do with bit shift (it's for programmers or people who know about it).

I took another look at the items you purchased.
It's definitely a device
ESP32:
"ESP32-WROOM-32 board"
I would think the "opendtu-generic_esp32.bin" should work.
 
It's a stupid question: Have you tried de-energizing the ESP32 to restart it? Sometimes that helps too.

**Reply by ghost** [reply]
It didn't work immediately for me either.
I initially had problems with the nrf24 module. My setup with display was on a breadboard.
I thought the nrf24 module was broken.
I then bought everything (except the display) with an openDTU board and soldered it in. Suddenly everything worked.
It seems that something was interfering with the free wiring.

**Reply by ghost** [reply]
Have you tried the seller's firmware?
The seller is the one where you got the wiring diagram.
[Website](https://waldy-smarthome.de/opendtu/)
[Webupdate](https://waldy-smarthome.de/opendtu/webflash/)
It's older, but he may have been testing it with it.

**Reply by FabienArr** [reply]
Yes i tried differents phones power supply, 
I have installed the last "opendtu-generic_esp32.bin". What is the difference between the factory and the generic ?
I will try again the old firmware.
I'm searching an openDTU board for cmt and not nrf, it's more difficult to find out, and all is in germany and 15€ of delivery for a board at 3€ is expansive.

**Reply by FabienArr** [reply]
I retried the seller's firmware without success.
I put a capacitor of 1uf, do you think it could be the problem ?
I found too, that i don't have sck but scl pin on my screen, is this the problem ?

**Reply by FabienArr** [reply]
I test the +5v and the GND of the screen, i have 4,7V
After i test the SCL and the SDA with GND and i get 3,2V. Is this OK ?

**Reply by FabienArr** [reply]
Hello,

Thanks for your replies.
Yes i have the CMT with a board and 8 pin header like that : 
![image](https://github.com/tbnobody/OpenDTU/assets/87081538/1e8841d7-75a5-4aa6-9b54-50b80c2b20d7)

For the voltage, a picture is better : 
![image](https://github.com/tbnobody/OpenDTU/assets/87081538/21ca8a4d-8f25-4af1-ac0f-f66afb70986a)
But for the 5V, there is only 2 pin for power component, the 3,3V is for the CMT module and the second of 5 v is for screen no ?

For the capacitor, it's just that i have only 1uf in stock. But the radio module works perfectly.

**Reply by FabienArr** [reply]
Hi,

I have a 32-pin ESP32 so the board will not be compatible.
I tried to put on D13 and D12 without success. Do you think i need to use another LCD ? or more an ESP ?
I found a french article on LCD : 0x7A(8bit) = 0x3C (7bit), the cms resistor is on the 0x7A so i need to use the 0x3C to communicate with it
https://passionelectronique.fr/ecran-oled-i2c-arduino/
Where i can find this information in openDTU ?
I think i will buy a screen and esp on ali.... for open dtu

Regards

**Comment by FabienArr** [comment]
Hello,

I think i will buy the 2 components.
I found a PCB card, do you think it's better than cable wiring ?

Regards

**Reply by FabienArr** [reply]
Hi,

I place the order. I will come back in 2 weeks.

Regards

**Reply by FabienArr** [reply]
Hi,

The screen is dead, the new one works perfectly.

Sorry and thanks.
